exporter/exporterhelper/common.go

```diff
@@ -56,99 +89,68 @@ func (req *baseRequest) OnProcessingFinished() {
 	}
 }
 
-// baseSettings represents all the options that users can configure.
-type baseSettings struct {
-	component.StartFunc
-	component.ShutdownFunc
-	consumerOptions []consumer.Option
-	TimeoutSettings
-	queue internal.ProducerConsumerQueue
-	RetrySettings
-	requestExporter bool
-	marshaler       internal.RequestMarshaler
-	unmarshaler     internal.RequestUnmarshaler
-}
-
-// newBaseSettings returns the baseSettings starting from the default and applying all configured options.
-// requestExporter indicates whether the base settings are for a new request exporter or not.
-// TODO: The first three arguments will be removed when the old exporter helpers will be updated to call the new ones.
-func newBaseSettings(requestExporter bool, marshaler internal.RequestMarshaler,
-	unmarshaler internal.RequestUnmarshaler, options ...Option) *baseSettings {
-	bs := &baseSettings{
-		requestExporter: requestExporter,
-		TimeoutSettings: NewDefaultTimeoutSettings(),
-		// TODO: Enable retry by default (call DefaultRetrySettings)
-		RetrySettings: RetrySettings{Enabled: false},
-		marshaler:     marshaler,
-		unmarshaler:   unmarshaler,
-	}
-
-	for _, op := range options {
-		op(bs)
-	}
-
-	return bs
-}
-
 // Option apply changes to baseSettings.
-type Option func(*baseSettings)
+type Option func(*baseExporter)
```

Comment:
Changing this signature should not be considered a breaking change since the argument is of unexported type, so users cannot implement it directly

Reply:
would suggest in a separate PR to change this to an interface with a private func. I think this is kind of the recommended new way of doing options.